src/cleanfigure.m

```diff
@@ -786,57 +716,60 @@ function movePointsCloser(meta, handle)
 end
 % =========================================================================
 function xNew = moveToBox(x, xRef, xLim, yLim)
-  % Takes a box defined by xlim, ylim, one point x and a reference point
-  % xRef.
-  % Returns the point xNew that sits on the line segment between x and xRef
-  % *and* on the box. If several such points exist, take the closest one
-  % to x.
+  % Takes a box defined by xlim, ylim, a vector of points x and a vector of 
+  % reference points xRef.
+  % Returns the vector of points xNew that sits on the line segment between 
+  % x and xRef *and* on the box. If several such points exist, take the 
+  % closest one to x.
+  n = size(x, 1);
 
   % Find out with which border the line x---xRef intersects, and determine
   % the smallest parameter alpha such that x + alpha*(xRef-x)
-  % sits on the boundary.
-  minAlpha = inf;
+  % sits on the boundary. Otherwise set Alpha to inf.
+  minAlpha = inf(n, 1);
+  
+  % Get the corner points
   [bottomLeft, topLeft, bottomRight, topRight] = corners(xLim, yLim);
-  %TODO: clean up duplicate code below, possibly store lambda in matrix
 
   % left boundary:
-  lambda = crossLines(x, xRef, bottomLeft, topLeft);
-  if 0.0 < lambda(2) && lambda(2) < 1.0 && abs(minAlpha) > abs(lambda(1))
-      minAlpha = lambda(1);
-  end
+  minAlpha = updateAlpha(x, xRef, bottomLeft, topLeft, minAlpha);
 
   % bottom boundary:
-  lambda = crossLines(x, xRef, bottomLeft, bottomRight);
-  if 0.0 < lambda(2) && lambda(2) < 1.0 && abs(minAlpha) > abs(lambda(1))
-      minAlpha = lambda(1);
-  end
+  minAlpha = updateAlpha(x, xRef, bottomLeft, bottomRight, minAlpha);
 
   % right boundary:
-  lambda = crossLines(x, xRef, bottomRight, topRight);
-  if 0.0 < lambda(2) && lambda(2) < 1.0 && abs(minAlpha) > abs(lambda(1))
-      minAlpha = lambda(1);
-  end
-
+  minAlpha = updateAlpha(x, xRef, bottomRight, topRight, minAlpha);
+  
   % top boundary:
-  lambda = crossLines(x, xRef, topLeft, topRight);
-  if 0.0 < lambda(2) && lambda(2) < 1.0 && abs(minAlpha) > abs(lambda(1))
-      minAlpha = lambda(1);
-  end
+  minAlpha = updateAlpha(x, xRef, topLeft, topRight, minAlpha);
 
-  % create the new point
-  xNew = x + minAlpha*(xRef-x);
+  % Create the new point
+  xNew = x + bsxfun(@times ,minAlpha, (xRef-x));
+end
+% =========================================================================
+function minAlpha = updateAlpha(X1, X2, X3, X4, minAlpha)
+  % Checks whether the segments X1--X2 and X3--X4 intersect. 
+  lambda = crossLines(X1, X2, X3, X4);
+  
+  % Check if lambda is in bounds and lambda1 large enough
+  id_Alpha           = 0.0 < lambda(:,2) & lambda(:,2) < 1.0 ...
+                     & abs(minAlpha) > abs(lambda(:,1));
```

Comment:
It is not really segmentsIntersect. The real lifting is crossLines. segmentsIntersect returns if the line crosses a border, whereas this actually checks where it crosses the border and if it is closer to the other borders.